### ISSUE TITLE
Feature request Multi Index pivot

### ISSUE THREAD

**Issue by kirialis** [pandas-dev/pandas#32129]
Hi, Would it be possible to implement pivot with a list of index ?

This will throw an error :
```python
df.pivot(index=['time', 'category'], columns='bar', values='baz')
```
But this does : (Found on Stack Overflow)
```python
def multiindex_pivot(df, index=None, columns=None, values=None):
    if index is None:
        names = list(df.index.names)
        df = df.reset_index()
    else:
        names = index
    list_index = df[names].values
    tuples_index = [tuple(i) for i in list_index] # hashable
    df = df.assign(tuples_index=tuples_index)
    df = df.pivot(index="tuples_index", columns=columns, values=values)
    tuples_index = df.index  # reduced
    index = pd.MultiIndex.from_tuples(tuples_index, names=names)
    df.index = index
    return df

df.pipe(multiindex_pivot, index=['time', 'category'], columns='bar', values='baz')
```

I just think it would be nice to have something directly available :)

**Comment by jreback** [comment]
this is already implemented for 1.1 if you’d like to find the issue / PR

**Comment by charlesdong1991** [comment]
indeed, @kirialis , your description of feature request seems like being implemented in #30928 , and original issue is #21425 , could you please try to merge master, and see if your issue resolved?

FYI, another PR to add annotations for `df.pivot` is underway #32197
